aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3422,16 +3427,40 @@ static void linalg_lstsq_out_info(
     TORCH_INTERNAL_ASSERT(singular_values.sizes().equals(singular_values_shape));
     TORCH_INTERNAL_ASSERT(singular_values.is_contiguous());
   }
-
   // 'input' is modified in-place so we need a column-major copy
   auto input_working_copy = copyBatchedColumnMajor(input);
-
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if(driver == "gelsd" || driver == "gelss") {
+    auto k = std::min(m, n);
+    auto U = at::zeros({input.size(0), m, k}, input.options());
+    auto Vh = at::zeros({input.size(0), k, n}, input.options());
+
+    svd_stub(input.device().type(),
+          input,
+          false, // you don't want the full SVD for least squares
+          true, // we need U, S, Vh for least squares
+          "gesvd",
```

Comment:
Not sure if the docs indirectly imply `gesvd` is "safer" than `gesvdj` because it says it'll fallabck to this . So for brevity, I'll leave it as this as a default. https://pytorch.org/docs/stable/generated/torch.linalg.svd.html

Reply:
`gesvd` is the right one to use here.